ibis/backends/base/sql/alchemy/query_builder.py

```diff
@@ -109,8 +108,14 @@ def _format_table(self, expr):
             backend._create_temp_view(view=result, definition=definition)
         elif isinstance(ref_op, ops.InMemoryTable):
             columns = _schema_to_sqlalchemy_columns(ref_op.schema)
-            rows = list(ref_op.data.itertuples(index=False))
-            result = sa.values(*columns).data(rows)
+
+            if self.context.compiler.cheap_in_memory_tables:
+                result = sa.table(ref_op.name, *columns)
```

Comment:
Is `ref_op.name` guaranteed to be unique? Where do these names come from?

Reply:
They are either generated or user provided. `_ibis_memtable_{a_nonzero_number}` is the template for the generated names.

Reply:
The names aren't guaranteed to be unique if the user passes `name`, but that's true of any other API that we provide that accepts an argument for naming a table.

If no name is passed, then there's potentially overlapping names if a table in the database is called `_ibis_memtable_{number}`. However, since such expressions would hash differently, they'll be unique and aliasing will take care of making sure things are unique except in 2 cases:

1. joining the two tables that have the same name
2. set operations on two tables with the same name

I specifically chose a prefix that I think will be extremely unlikely to collide with anyone's table names. An alternative would be to guarantee uniqueness by using using a UUID, but then the table name is hard to read IMO.

Reply:
No, what you have here is fine - I was just curious how this worked. Thanks!